internal/resource/manage/timeout/timeout_test.go

```diff
@@ -0,0 +1,73 @@
+package timeout_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/slok/kahoy/internal/model"
+	"github.com/slok/kahoy/internal/resource/manage/managemock"
+	"github.com/slok/kahoy/internal/resource/manage/timeout"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTimeoutManagerApply(t *testing.T) {
+	tests := map[string]struct {
+		config    timeout.TimeoutManagerConfig
+		resources []model.Resource
+		mock      func(mrm *managemock.ResourceManager)
+		expErr    bool
+	}{
+		"Basic initialization should not fail.": {
+			mock: func(mrm *managemock.ResourceManager) {
+				mrm.On("Apply", mock.Anything, mock.Anything).Once().Return(nil)
+			},
+		},
+
+		"If Apply takes longer than timeout, apply should fail.": {
+			config: timeout.TimeoutManagerConfig{
+				Timeout: 1 * time.Second,
```

Comment:
is 1 second too much? is it even a good idea? @slok 

The overall idea of this test is that I configure a low timeout so that the manager timeouts before the mock manager is able to finish the apply.

Reply:
I resolved this issue with a custom manager implementation that handles context cancelation.